File: src/main/java/dev/mazurkiewicz/quizer/config/QuizerProperties.java
package dev.mazurkiewicz.quizer.config;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

@AllArgsConstructor
@ConfigurationProperties(prefix = "dev.mazurkiewicz.quizer")
@Getter
@NoArgsConstructor
@Setter
public class QuizerProperties {
    private Integer examTimeInSeconds;
    private String pdfName;
    private String questionsImgFolder;
    private String resourcesFolder;
    private Integer testQuestionsNumber;
}

File: src/main/kotlin/pl/quizwiz/config/QuizerProperties.kt
package pl.quizwiz.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "quizer")
data class QuizerProperties(
    val quizTimeInSeconds: Int,
    val pdfName: String,
    val questionsImgFolder: String,
    val resourcesFolder: String,
    val quizQuestionNumber: Int,
    val quizPassThreshold: Int
)